packages/typespec-ts/src/transform/transformParameters.ts

```diff
@@ -256,122 +260,101 @@ function transformBodyParameters(
   if (!bodyType) {
     return;
   }
-  const { hasBinaryContent } = getBodyDetail(dpgContext, bodyType, headers);
-
-  return transformNormalBody(
+  return transformRequestBody(
     dpgContext,
     bodyType,
     parameters,
     importedModels,
-    headers,
-    hasBinaryContent
+    headers
   );
 }
 
-function transformNormalBody(
+function transformRequestBody(
   dpgContext: SdkContext,
   bodyType: Type,
   parameters: HttpOperationParameters,
   importedModels: Set<string>,
-  headers: ParameterMetadata[],
-  hasBinaryContent: boolean
+  headers: ParameterMetadata[]
 ) {
-  const descriptions = extractDescriptionsFromBody(
-    dpgContext,
-    bodyType,
-    parameters
-  );
-  if (hasBinaryContent) {
-    descriptions.push("Value may contain any sequence of octets");
-  }
-  const type = extractNameFromTypeSpecType(
-    dpgContext,
-    bodyType,
+  const contentTypes = extractMediaTypes(parameters.body?.contentTypes ?? []);
+  const schema = getSchemaForType(dpgContext, bodyType, {
+    mediaTypes: contentTypes,
+    isRequestBody: true,
+    usage: [SchemaContext.Input, SchemaContext.Exception]
+  });
+
+  const descriptions = getBodyDescriptions(dpgContext, schema, parameters);
+  const type = getRequestBodyType(
+    schema,
     [SchemaContext.Input],
     importedModels,
     headers
   );
-  let schema = getSchemaForType(dpgContext, bodyType);
-  let overrideType = undefined;
-  if (hasBinaryContent) {
-    schema = enrichBinaryTypeInBody(schema);
-    overrideType = schema.typeName;
-  }
+
   return {
     isPartialBody: false,
+    hasFileTypeIncluded: isFileUploadCase(),
     body: [
       {
         properties: schema.properties,
         typeName: schema.name,
         name: "body",
-        type: overrideType ?? type,
+        type: type,
         required: parameters?.bodyParameter?.optional === false,
         description: descriptions.join("\n\n"),
         oriSchema: schema
       }
     ]
   };
-}
+  function isFileUploadCase() {
```

Comment:
Shall we call this isMultpartFileUpload?
Do we have cases that is also file uploading but not using multpart?

Reply:
Sure we could change to isMultpartFileUpload. Yeah, i think there is no other file case yet.